### PR TITLE
doc: add hugegraph server setup guide for IDEA

### PR DESCRIPTION
**Pull request by VGalaxies**
The [Quick Start](https://hugegraph.apache.org/docs/quickstart/hugegraph-server/) section provides instructions on how to start and stop HugeGraphServer using scripts. However, for developers, this method may not be convenient as it is not integrated with IDE tools and can be difficult to debug. 

So I add a guide to explain how to run and debug HugeGraphServer on the Linux platform using IntelliJ IDEA.

I have read the CLA Document and I hereby sign the CLA

### CODE REVIEW

File: content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md
Comment (imbajin) [inline content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md:1]: 可以复制这段话到 GPT/New Bing 中优化一下, 部分描述还可以优化一下

File: content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+title: "如何在 IDEA 中搭建 HugeGraph-Server 开发环境"
+linkTitle: "如何在 IDEA 中搭建 HugeGraph-Server 开发环境"
+weight: 4
+---
+
+> 注意：下述配置仅供参考，基于[这个版本](https://github.com/apache/incubator-hugegraph/commit/a946ad1de4e8f922251a5241ffc957c33379677f)，在 Linux 和 macOS 平台下进行了测试。
+
+### 背景
+
+在 [Quick Start](/docs/quickstart/hugegraph-server/) 部分中已经给出了使用脚本启停 HugeGraphServer 的流程，但是对于开发者而言，这种方式没有与 IDE 工具集成，使用脚本会较为繁琐，调试起来也比较麻烦。下面以 Linux 平台为例，介绍使用 IntelliJ IDEA 运行与调试 HugeGraph-Server 的流程。
```
Comment (imbajin) [inline content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md:11]: 可以简化下描述

File: content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+本地启动的核心与**脚本启动**是一样的：
+
+1. 初始化数据库后端, 执行 `InitStore` 类初始化图
+2. 启动 HugeGraphServer，执行 `HugeGraphServer` 类加载初始化的图信息启动
+
+在进行下述流程前，请确保已经 clone 了 HugeGraph 的源代码，并且已经配置了 JDK 11 等开发环境。
+
+```
```
Comment (imbajin) [inline content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md:20]: add `bash` for code snippet

File: content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+cp -r hugegraph-dist/src/assembly/static/scripts hugegraph-dist/src/assembly/static/conf <path-to-your-directory>
+```
+
+#### 2. `InitStore` 运行配置
+
+首先，需要在拷贝的文件中配置数据库后端，这里以 RocksDB 为例，打开 `<path-to-your-directory>/conf/graphs/hugegraph.properties`，配置如下：
+
+```
```
Comment (imbajin) [inline content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md:39]: add `properties` tag for it

File: content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+cd hugegraph
+```
+
+### 步骤
+
+#### 1. 配置文件拷贝
+
+为了避免配置文件的变动影响 Git 的追踪，建议将所需的配置文件拷贝到一个单独的文件夹中：
+
+```
```
Comment (imbajin) [inline content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md:31]: same

File: content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+2023-06-05 00:43:39 [main] [INFO] o.a.h.b.c.CacheManager - Init RamCache for 'users_pwd-hugegraph' with capacity 10240
+2023-06-05 00:43:39 [main] [INFO] o.a.h.b.c.CacheManager - Init RamCache for 'token-hugegraph' with capacity 10240
+2023-06-05 00:43:39 [main] [INFO] o.a.h.b.s.r.RocksDBStore - Write down the backend version: 1.11
+2023-06-05 00:43:39 [main] [INFO] o.a.h.StandardHugeGraph - Graph 'hugegraph' has been initialized
+2023-06-05 00:43:39 [main] [INFO] o.a.h.StandardHugeGraph - Close graph standardhugegraph[hugegraph]
+2023-06-05 00:43:39 [db-open-1] [INFO] o.a.h.b.s.r.RocksDBStore - Opening RocksDB with data path: ./m
+2023-06-05 00:43:39 [db-open-1] [INFO] o.a.h.b.s.r.RocksDBStore - Opening RocksDB with data path: ./s
+2023-06-05 00:43:39 [db-open-1] [INFO] o.a.h.b.s.r.RocksDBStore - Opening RocksDB with data path: ./g
+2023-06-05 00:43:39 [main] [INFO] o.a.h.HugeFactory - HugeFactory shutdown
+2023-06-05 00:43:39 [hugegraph-shutdown] [INFO] o.a.h.HugeFactory - HugeGraph is shutting down
```
Comment (imbajin) [inline content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md:78]: 1. add `java` for the code snippet
2. could simplify the log (like ignore the cache log with `...`)

File: content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+此时可以在 Debugger 中查看详细的变量信息。
+
+### 可能出现的问题
+
+#### java: package sun.misc does not exist
+
+原因可能因为使用 Java 11 编译时触发了交叉编译，导致项目中使用的 `sun.misc.Unsafe` 找不到符号，有两种解决方案：
+
+1. 设置 Project SDK 为 8，或者
+2. 在 IDEA 的 `Preference` 中找到 `Java Compiler` 面板，然后关闭 `--release` 选项
```
Comment (imbajin) [inline content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md:134]: 交换次序, 应优先使用第二个方式, 而不是降级 `Java SDK` (因为下一个大版本可能取消对 java 8 支持)

File: content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+配置完成后运行，如果看到下述日志，代表 `HugeGraphServer` 已经成功启动：
+
+```
+......
+2023-06-05 00:51:56 [gremlin-server-boss-1] [INFO] o.a.t.g.s.GremlinServer - Gremlin Server configured with worker thread pool of 1, gremlin pool of 8 and boss thread pool of 1.
+2023-06-05 00:51:56 [gremlin-server-boss-1] [INFO] o.a.t.g.s.GremlinServer - Channel started at port 8182.
+```
+
+#### 4. 调试 `HugeGraphServer`
```
Comment (imbajin) [inline content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md:97]: 最后可加上 (可选/.示例) 表明此步是一个参考例子

File: content/en/docs/contribution-guidelines/hugegraph-server-idea-setup.md
Comment (imbajin) [inline content/en/docs/contribution-guidelines/hugegraph-server-idea-setup.md:1]: update the EN doc (together) until the CN doc finished

File: content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+                   @QueryParam("label") String label,
+                   @QueryParam("properties") String properties,
+                   @QueryParam("keep_start_p")
+                   @DefaultValue("false") boolean keepStartP,
+                   @QueryParam("offset") @DefaultValue("0") long offset,
+                   @QueryParam("page") String page,
+                   @QueryParam("limit") @DefaultValue("100") long limit) {
+    LOG.debug("Graph [{}] query vertices by label: {}, properties: {}, " +
+              "offset: {}, page: {}, limit: {}",
+              graph, label, properties, offset, page, limit);
```
Comment (imbajin) [inline content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md:118]: could also simplify it: reduce the params & ignore the log line

File: content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+
+为了避免配置文件的更改影响 Git 的追踪，建议将所需的配置文件拷贝到一个单独的文件夹中：
+
+```bash
+cp -r hugegraph-dist/src/assembly/static/scripts hugegraph-dist/src/assembly/static/conf path-to-your-directory
+```
+
+#### 2. `InitStore` 类初始化图
+
+首先，需要在配置文件中配置数据库后端。以 RocksDB 为例，在 `path-to/conf/graphs/hugegraph.properties` 文件中进行以下配置：
```
Comment (imbajin) [inline content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md:36]: `<>` may hit the `HTML` tag and influence the UI, so use `xx` for it or just remove the `<>`

image

**Comment by imbajin** [comment]
@liuxiaocs7 @Radeity @msgui could also take a look for it (Review CN doc only)

File: content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+
+此时，可以在调试器中查看详细的变量信息。
+
+### 可能遇到的问题
+
+*** java: package sun.misc does not exist ***
+
+原因可能是在使用 Java 11 编译时触发了交叉编译，导致项目中使用的 `sun.misc.Unsafe` 找不到符号。有两种解决方案可供选择：
+
+1. 在 IntelliJ IDEA 的 `Preferences` 中找到 `Java Compiler` 面板，然后关闭 `--release` 选项 (推荐)
```
Comment (Radeity) [inline content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md:109]:
```suggestion
1. 在 IntelliJ IDEA 的 `Preferences/Settings` 中找到 `Java Compiler` 面板，然后关闭 `--release` 选项 (推荐)
```

File: content/en/docs/contribution-guidelines/hugegraph-server-idea-setup.md [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+
+At this point, you can view detailed variable information in the debugger.
+
+### Possible Issues
+
+#### java: package sun.misc does not exist
+
+The reason may be that cross-compilation is triggered when using Java 11 to compile, causing the symbol of `sun.misc.Unsafe` used in the project to not be found. There are two possible solutions:
+
+1. In IntelliJ IDEA, go to `Preferences` and find the `Java Compiler` panel. Then, disable the `--release` option, or
```
Comment (Radeity) [inline content/en/docs/contribution-guidelines/hugegraph-server-idea-setup.md:115]:
```suggestion
1. In IntelliJ IDEA, go to `Preferences/Settings` and find the `Java Compiler` panel. Then, disable the `--release` option, or
```

File: content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md (hunk 10 above)
Comment (VGalaxies) [reply content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md:109]: Thanks, I'll add it soon.

File: content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md (hunk 9 above)
Comment (VGalaxies) [reply content/cn/docs/contribution-guidelines/hugegraph-server-idea-setup.md:36]: Sorry I overlooked this case, I will remove the rest `<>` accordingly.

**Review by liuxiaocs7 — DISMISSED** [review]
+1 after fix above comment

**Review by imbajin — APPROVED** [review]
LGTM

**Review by coderzc — APPROVED** [review]

**Pull request merged**